pkg/probe/http/http_test.go

```diff
@@ -158,13 +158,21 @@ func TestHTTPProbeChecker(t *testing.T) {
 			handler:    headerCounterHandler,
 			reqHeaders: http.Header{},
 			health:     probe.Success,
-			accBody:    "4",
+			accBody:    "3",
 		},
 		{
 			handler:    headerKeysNamesHandler,
 			reqHeaders: http.Header{},
 			health:     probe.Success,
-			accBody:    "Accept\nAccept-Encoding\nConnection\nUser-Agent",
+			accBody:    "Accept\nConnection\nUser-Agent",
+		},
+		{
+			handler: headerEchoHandler,
+			reqHeaders: http.Header{
+				"Accept-Encoding": {"gzip"},
```

Comment:
what will happen with the empty string for the `Accept-Encoding`? I wonder for the behavor before this change and after. Would any users already setting it to empty experience a different behavior?

Reply:
https://tools.ietf.org/html/rfc7231#section-5.3.4

`
If the representation has no content-coding, then it is  acceptable by default unless specifically excluded by the Accept-Encoding field stating either "identity;q=0" or "*;q=0" without a more specific entry for "identity"
`

My understanding is that the server should translate an "Accept-Encoding: " into "Accept-Encoding: identity"

So I don't anticipate any breaking change in this situation

Reply:
I was mostly asking from the client perspective. If empty string was set for `"Accept-Encoding"` in yaml, will it translate to no header or to empty string header before and after this PR?

Reply:
ah, it should translate to empty string as empty string is a valid "Accept-Encoding". I will add a test

Reply:
done 👍  ty